flaml/ml.py

```diff
@@ -181,7 +181,10 @@ def metric_loss_score(
                         for (prediction, label) in zip(y_predict, y_true)
                     ]
                     y_pred = [
-                        [labels[p] for (p, l) in each_list]
+                        [
+                            labels[p] if p in range(len(labels)) else -1

```

Comment:
Please use a comparison expression instead of "in" which is much more efficient.
Also, remember to compute `len(labels)` out of the loop to be efficient.